packages/runtime/datastore/src/localChannelStorageService.ts

```diff
@@ -57,4 +66,29 @@ export class LocalChannelStorageService implements IChannelStorageService {
 
         return undefined;
     }
+
+    private readStringSyncInternal(path: string, tree: ITree): string | undefined {
```

Comment:
Instead of duplicating code, what you can do is change readSyncInternal to return IBlob. And then callers to do last mile translation to needed format.

Reply:
Or alternatively, implement readBlob() API on this object (leveraging existing readSyncInternal functionality), and then make readString() implemented by calling it. 
Note that we are attempting to get rid of read() API, so you that's another way not to duplicate code - you can just repurpose existing readSyncInternal() to work with decoded strings.